Import names from module

xealits opened the pull request:
The `importNames` and `importNames_sync` procedures import objects from the given module into current global space (`priv->globals`).

The definitions:

    void QPython::importNames(QString name, QVariant args, QJSValue callback)
    bool QPython::importNames_sync(QString module_name, QVariant args)

where `args` is the list with names of objects to import.

The procedures follow `importModule` code scheme.
`importNames_sync` starts with copy of `importModule_sync`,
but does not add the imported module to the `dict` `priv->globals`.
It creates a temporary dictionary filled with `priv->globals` and the module.
The objects are obtained by running `PyRun_String` on `'module_name.obj_name'` strings
in the loop over `obj_names` with the temp dictionary as global scope.
On success the object is inserted into `priv->globals`.

Following [the issue 53](https://github.com/thp/pyotherside/issues/53)

thp commented on src/qpython.cpp:
```diff
@@ -127,6 +129,103 @@ QPython::addImportPath(QString path)
+
+    ENSURE_GIL_STATE;
+
+    bool use_api_10 = (api_version_major == 1 && api_version_minor == 0);
```
This is new API so it didn't exist with 1.0, right? No need to differentiate between APIs.

thp commented on src/qpython.cpp:
```diff
@@ -127,6 +129,103 @@ QPython::addImportPath(QString path)
+        // however, further PyRun_String is used to get the object
+        full_name = module_name + "." + obj_name;
+        utf8bytes = full_name.toUtf8();
+        PyObject *res = PyRun_String(utf8bytes.constData(),
```
Please use `PyObject_GetAttr` ([docs here](https://docs.python.org/3.4/c-api/object.html#c.PyObject_GetAttr)) for getting an object from a module, not by building and evaluating strings.

thp replied:
Or actually what you want is probably `PyObject_GetAttrString` (which takes a `char *`).

thp commented:
Have you had time to address the comments?

xealits commented:
Sorry, not yet. I'll try to do so this week.

xealits commented:
Ok, it should be better now!

thp commented on src/qpython.h:
```diff
@@ -131,6 +131,61 @@ class QPython : public QObject {
+         * \code
+         * Python {
+         *     Component.onCompleted: {
+         *         importNames('os', ['path'] function() {
```
Missing a `,` between `]` and `function`. Also, it returns a boolean that the function could get as parameter:

```
importNames('os', ['path'], function (success) {
    if (success) {
        // You can use the "path" submodule here
    } else {
        console.log('Importing failed')
    }
})
```

thp commented:
Looks good; can you fix the last remaining documentation issue and then squash the commits, so it can be merged? Thanks :)

xealits commented:
I fixed the documentation and squashed the commits.
But there was an error with importing from packaged modules `from x.y.z import a, b, c`.
I corrected it by using `PyImport_ImportModule` for module import.
Now it should be good, could you check?

thp commented:
Perfect now, thanks :) Merging...
